src/srx_caproto_iocs/base.py

```diff
@@ -128,9 +128,12 @@ async def stage(self, instance, value):
                 return False
 
             file_name = self.file_name.value
-            uid = "" if "{uid" not in file_name else str(uuid.uuid4())
+            uid = "" if not any(
+                s in file_name for s in ("{uid", "{suid")
+            ) else str(uuid.uuid4())
+            uid = "" if "{uid" not in file_name or "{suid" not in file_name else str(uuid.uuid4())
```

Comment:
I think it should be `and` here.
```suggestion
            uid = "" if "{uid" not in file_name and "{suid" not in file_name else str(uuid.uuid4())
```

Reply:
> I think it should be `and` here.

Yes, you are right. I meant to remove that line and keep L131:134 because I felt it is more readable.

But I found that the first arg does not actually get evaluated before the condition, so we could simplify the logic a bit.
```py
In [4]: def foo():
   ...:     print("foo is called")
   ...:     return "foo"
   ...: 

In [5]: s = foo() if False else "bar"

In [6]: s
Out[6]: 'bar'

```

Reply:
| `uid` | `suid` | `or` logic | `and` logic |
| ---| -----| --------| ------------|
| used | used  | `str(uuid.uuid4())` | `str(uuid.uuid4())` |
| used | unused | `""` | `str(uuid.uuid4())` |
| unused | used | `""` | `str(uuid.uuid4())` |
| unused | unused | `""` | `""` |